docs/howtos/DataPlane Generation - DPG/08methodInputs.mdx

```diff
@@ -0,0 +1,456 @@
+
+```csharp
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+```
+
+</TabItem>
+<TabItem value="go" label="Go" >
+
+```go
+```
+
+</TabItem>
+</Tabs>
+
+## Spread
+
+### Alias
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+alias User = {
+  firstName: string,
+  lastName: string,
+};
+
+op upload(...User): void;
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+For Python, we will also generate the overloads described in the Http Post section, but omitting for brevity
+
+```python
+def upload(first_name: str, last_name: str) -> None:
+  ...
+```
+
+</TabItem>
+<TabItem value="csharp" label="CSharp" >
+
+```csharp
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+```
+
+</TabItem>
+<TabItem value="go" label="Go" >
+
+```go
+```
+
+</TabItem>
+</Tabs>
+
+### Alias with HTTP Parameters
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+alias User = {
+  @path id: string,
+  firstName: string,
+  lastName: string,
+};
+
+op upload(...User): void;
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+```python
+def upload(id: str, first_name: str, last_name: str) -> None:
+  ...
+```
+
+</TabItem>
+<TabItem value="csharp" label="CSharp" >
+
+```csharp
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+```
+
+</TabItem>
+<TabItem value="go" label="Go" >
+
+```go
+```
+
+</TabItem>
+</Tabs>
+
+### Named model
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+model User {
+  firstName: string;
+  lastName: string;
+}
+
+op upload(...User): void;
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+```python
+class User:
+  first_name: str
+  last_name: str
+
+def upload(user: User) -> None:
+  ...
+```
+
+</TabItem>
+<TabItem value="csharp" label="CSharp" >
+
+```csharp
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+```
+
+</TabItem>
+<TabItem value="go" label="Go" >
+
+```go
+```
+
+</TabItem>
+</Tabs> 
+
+### Model with @body Property
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+model User {
+  firstName: string;
+  lastName: string;
+}
+
+model UserRequest {
+  @body user: User;
+}
+
+op upload(...UserRequest): void;
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+For Python, we will also generate the overloads described in the Http Post section, but omitting for brevity
+
+```python
+class User:
+  first_name: str
+  last_name: str
+
+class UserRequest:
+  user: User
+
+def upload(userRequest: UserRequest) -> None:
+  ...
```

Comment:
Not sure if Python code is correct.

If `@body user: User`, the request body is the `User` model.

Reply:
@iscai-msft The code seems not correct and here is the generated code with latest typespec-emitter:
![image](https://github.com/Azure/typespec-azure/assets/70930885/fc2214ce-a881-4d3a-b5b4-146b69885904)